### ISSUE TITLE
BuddyAllocator::pairs Grows Without Bound with Steady Allocation/Deallocation

### ISSUE THREAD

**Issue by cwfitzgerald** [zakarumych/gpu-alloc#55]
## Related

This was initially reported as https://github.com/BVE-Reborn/rend3/issues/167.

## Description

rend3, in the course of normal operation, allocates/deallocates 5 buffers per frame from wgpu. If the camera does not move, these buffers will be the exact same size.

Especially when running at high frame rates, this causes `BuddyAllocator::pairs` to slowly increase in size to the tune of 1MB every couple seconds (depending on framerate).

This definitely feels unexpected.

## More Information

If you need more information, please reach out, I'm not sure exactly what would be of most help.

## Repro

This repros on vulkan on windows.

```
git clone https://github.com/BVE-Reborn/rend3.git
cd rend3
cargo run --bin scene-viewer --release -- -m gpu
```

Watch the memory usage of the application slowly rise.

**Comment by zakarumych** [comment]
RAM or GPU memory?

**Comment by zakarumych** [comment]
I've published new patch version 0.5.1 with the fix.

**Comment by cwfitzgerald** [comment]
Great! Thank you so much!
